bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/ScalarWebClient.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.sony.internal.scalarweb;
+
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.sony.internal.scalarweb.models.ScalarWebService;
+
+/**
+ * This class represents a web scalar method definition that can be called
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+@NonNullByDefault
+public class ScalarWebClient implements AutoCloseable {
+    /** The device manager */
+    private final ScalarWebDeviceManager deviceManager;
+
+    /**
+     * Instantiates a new scalar web state.
+     *
+     * @param deviceManager the non-null device manager
+     */
+    public ScalarWebClient(final ScalarWebDeviceManager deviceManager) {
+        Objects.requireNonNull(deviceManager, "deviceManager cannot be null");
+
+        this.deviceManager = deviceManager;
+    }
+
+    /**
+     * Gets the device manager
+     *
+     * @return the non-null device manager
+     */
+    public ScalarWebDeviceManager getDevice() {
```

Comment:
Got to remember - alot of this has gone through iterations and ScalarWebClient use to be a huge class.  Since the rewritings - the functionality has dwindled and it looks like this is just a facade now.  I'm going to change ScalarWebClient to an interface (although the getDevice method is kinda weird now) and wow the comment on this class was way off!